src/main/java/org/openpnp/machine/reference/ReferenceNozzleTipCalibration.java

```diff
@@ -433,6 +432,25 @@ public Location getCameraOffset() {
     @Commit
     public void commit() {
         angleIncrement = null;
+
+        // We need to invoke this even later than the commit(), as the Configuration is not yet ready.
+        SwingUtilities.invokeLater(() -> {
```

Comment:
Hi @vonnieda 

Please review this use of `invokeLater()`. I got a null pointer exception when adding the listeners immediately. I'm still fresh with java :)

Thanks,
_Mark

Reply:
Hi @markmaker, in general, the UI should listen to the model, not the other way around. The reason for this is that in some cases the UI will not exist - unit tests being the primary one. When this code is running as a unit test it will fail because SwingUtilities will not have an event thread, and MainFrame.get() will return null.

I'm not sure I'm following what the listener is intended to do exactly, but it looks like the goal here is to have the wizard update the calibration status when the selected tool changes. If that's the case then simply have the wizard add the listener from it's constructor. At that time the UI is guaranteed to exist, and the configuration is guaranteed to have been loaded.

Other than that, I'm good with the rest of the PR.

Reply:
Hi @vonnieda 

Thanks for reviewing this. I made another proposition, see the commit. 

My conclusion was to move most of the `"runoutCompensationInformation"` stuff from `ReferenceNozzleTipCalibration` to `ReferenceNozzleTipCalibrationWizard`. I renamed it "calibrationStatus". Now the listeners work model-to-UI and UI-to-UI as is appropriate, I guess. 

Note, "model" is meant as in "model view control" here, **not** as in "org.openpnp.model". 

## Why the move?

The `"runoutCompensationInformation"` that needs updating was a property of the `ReferenceNozzleTipCalibration` i.e. a model property, not UI. When I moved the `addPropertyChangeListener()` to `ReferenceNozzleTipCalibrationWizard` (as you suggested) it felt wrong. I was wiring a model-to-model change notification through the GUI. If the same Wizard were to be created multiple times, the same listeners would also added multiple times (don't know if this can ever happen in OpenPNP but it certainly feels wrong). By moving the property to the Wizard, this is fixed. This also nicely sorted things out with `getUiCalibrationNozzle()` that is no longer needed on the model side but only in the UI where it belongs. 


_Mark